src/ert/_c_wrappers/enkf/enkf_fs.py

```diff
@@ -135,6 +141,77 @@ def realizationList(self, state: "RealizationStateEnum") -> "IntVector":
         state_map = self.getStateMap()
         return state_map.realizationList(state)
 
+    def _has_parameters(self) -> bool:
+        """
+        Checks if a parameter folder has been created
+        """
+        for path in self.mount_point.iterdir():
+            if "gen-kw" in str(path):
+                return True
+
+    def save_gen_kw(
+        self,
+        parameter_name: str,
+        parameter_keys: List[str],
+        realization: int,
+        data: npt.ArrayLike,
+    ) -> None:
+        output_path = self.mount_point / f"gen-kw-{realization}"
+        Path.mkdir(output_path, exist_ok=True)
+
+        np.save(output_path / parameter_name, data)
+        with open(output_path / f"{parameter_name}-keys", "w") as f:
+            f.write("\n".join(parameter_keys))
+
+        self.getStateMap().update_matching(
+            realization,
+            RealizationStateEnum.STATE_UNDEFINED,
+            RealizationStateEnum.STATE_INITIALIZED,
+        )
+
+    def _load_gen_kw_realization(
+        self, key: str, realization: int
+    ) -> Tuple["npt.NDArray[np.double]", List[str]]:
+        input_path = self.mount_point / f"gen-kw-{realization}"
+        if not input_path.exists():
+            raise KeyError(f"Unable to load GEN_KW for key: {key}")
+
+        np_data = np.load(input_path / f"{key}.npy")
+        with open(input_path / f"{key}-keys", "r") as f:
+            keys = [k.strip() for k in f.readlines()]
+
+        return np_data, keys
+
+    def load_gen_kw_as_dict(
```

Comment:
Do we need both this and `load_gen_kw`?

Reply:
load_gen_kw is used in es_update

Reply:
Could it be refactored to use the same API, or is it very different?

Reply:
it is quite different, but I think a reiteration of the API should be done when we have all the data